plugins/modules/aci_access_port_to_interface_policy_leaf_profile.py

```diff
@@ -382,16 +415,32 @@ def main():
 
     # Add infraRsAccBaseGrp only when policy_group was defined
     if policy_group is not None:
-        child_configs.append(
-            dict(
-                infraRsAccBaseGrp=dict(
-                    attributes=dict(
-                        tDn=INTERFACE_TYPE_MAPPING[interface_type].format(policy_group),
-                    ),
-                ),
-            )
+        if interface_type == "fex_profile" and type_profile == "fex":
+            module.fail_json(msg="Invalid Configuration - interface_type fex_profile can not be configured with a profile of type fex")
+        elif interface_type == "fex_profile" and fex_profile is not None:
+            infra_rs_acc_base_grp_t_dn = INTERFACE_TYPE_MAPPING[interface_type].format(fex_profile, policy_group)
+        elif interface_type == "fex_profile" and fex_profile is None:
```

Comment:
Since all the conditions are AND could this me merged into one condition?
if interface_type == "fex_profile" : 
   if type_profile == "fex": .......
   if fex_profile is not None: ...
else:
     infra_rs_acc_base_grp_t_dn = INTERFACE_TYPE_MAPPING[interface_type].format(policy_group)